components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementService.java

```diff
@@ -73,7 +73,18 @@
     /**
      * Method to get the count of all types of devices.
      * @return device count
-     * @throws DeviceManagementException
+     * @throws DeviceManagementException If some unusual behaviour is observed while counting
+     * the devices
      */
     int getDeviceCount() throws DeviceManagementException;
+
+    /**
+     * Method to get the list of devices that matches with the given device name.
+     *
+     * @param deviceName    name of the device
+     * @return List of devices that matches with the given device name.
+     * @throws DeviceManagementException If some unusual behaviour is observed while fetching the
+     * device list
+     */
+    List<Device> getDevicesByName(String deviceName, int tenantId) throws DeviceManagementException;
```

Comment:
The second parameter is removed in commit [1], hence nothing will not be changed.
https://github.com/wso2/carbon-device-mgt/commit/60960d80b4cfa1e7bf16c3fffb10f9295e2185ae